analysis/src/cratesio.rs

```diff
@@ -91,3 +91,26 @@ impl CratesioAnalyzer {
     }
 
 }
+
+#[cfg(test)]
+mod tests{
+    use super::*;
+
+    #[test]
+    fn test_unhosted_crate_name(){
+        let name = "::this crate does not exist::";
+        let cratesio_report = CratesioAnalyzer.analyze_cratesio(name).unwrap();
+        assert_eq!(cratesio_report.is_hosted, false);
+    }
+
+    #[test]
+    fn test_hosted_crate_name(){
+        let name = "guppy"; //BAD: as the test depends on guppy remaining hosted
```

Comment:
This isn't really "Bad" because crates cannot be removed from crates.io

Reply:
Though if you want to be even more future proof, use libc or some other extremely popular and stable crate.